src/router_core/transfer.c

```diff
@@ -745,6 +742,9 @@ static void qdr_link_deliver_CT(qdr_core_t *core, qdr_action_t *action, bool dis
 
     link->total_deliveries++;
 
+    if (link->link_type == QD_LINK_ENDPOINT)
```

Comment:
Should we also move the check for fallback here as well:

... ENDPOINT && !dlv_link->fallback)  ???